hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Path.java

```diff
@@ -199,8 +199,15 @@ public Path(String pathString) throws IllegalArgumentException {
     int start = 0;
 
     // parse uri scheme, if any
-    int colon = pathString.indexOf(':');
-    int slash = pathString.indexOf('/');
+    int colon = -1;
+    int slash = -1;
+    if (StringUtils.countMatches(pathString, ":") > 2) {
+      colon = pathString.indexOf(":/");
```

Comment:
Looks "slash = pathString.indexOf('/')" line is common in both If Else loops, can we bring this line out of If Else ?